spring-web/src/main/java/org/springframework/web/bind/support/WebExchangeDataBinder.java

```diff
@@ -87,6 +89,13 @@ public Mono<Void> construct(ServerWebExchange exchange) {
 				.then();
 	}
 
+	@Override
+	protected boolean shouldConstructArgument(MethodParameter param) {
+		Class<?> type = param.nestedIfOptional().getNestedParameterType();
+		return (super.shouldConstructArgument(param) &&
+				!MultipartFile.class.isAssignableFrom(type) && !Part.class.isAssignableFrom(type));
+	}
```

Comment:
This is a necessary change indeed for `Part` but `MultipartFile` is not used in WebFlux, so we can leave that out.